hashring/rbtree.go

```diff
@@ -261,28 +261,28 @@ func (n *redBlackNode) search(key keytype) (valuetype, bool) {
 	return nil, false
 }
 
-// walk Walks the tree in sorted order invoking `walker` for each node. If the
-// walker function returns `false`, walking is stopped and no more nodes will be
+// traverseUntil traverses the nodes in the tree in-order invoking `traverse` for each node. If the
+// traverse function returns `false`, traversal is stopped and no more nodes will be
 // visited. Returns `true` if all nodes are visited; `false` if not.
-func (n *redBlackNode) walk(walker func(*redBlackNode) bool) bool {
+func (n *redBlackNode) traverseUntil(traverse func(*redBlackNode) bool) bool {
```

Comment:
I still think this function is overloaded and has too many use cases but I'm cool with leaving it as is. However, this should be really called traverseWhile. "Until" indicates that the traversal should stop when the condition becomes true. "Do this until I give a signal". That means you stop when "I give a signal" is true. 

"Do this *while* I give you the signal" indicates that you should stop when "I give you the signal" is false. Which is what this function does. Does that make sense?

I also think the function should be called condition to make that more clear.